substrate/frame/contracts/uapi/src/api.rs

```diff
@@ -0,0 +1,759 @@
+	///
+	/// Writes the public key into the given output buffer.
+	/// Assumes the secp256k1 curve.
+	///
+	/// # Parameters
+	///
+	/// - `signature`: The signature bytes.
+	/// - `message_hash`: The message hash bytes.
+	/// - `output`: A reference to the output data buffer to write the public key.
+	///
+	/// # Errors
+	///
+	/// - [EcdsaRecoveryFailed][`crate::ReturnErrorCode::EcdsaRecoveryFailed]
+	fn ecdsa_recover(
+		signature: &[u8; 65],
+		message_hash: &[u8; 32],
+		output: &mut [u8; 33],
+	) -> Result;
+
+	/// Calculates Ethereum address from the ECDSA compressed public key and stores
+	/// it into the supplied buffer.
+	///
+	/// # Parameters
+	///
+	/// - `pubkey`: The public key bytes.
+	/// - `output`: A reference to the output data buffer to write the address.
+	///
+	/// # Errors
+	///
+	/// - [EcdsaRecoveryFailed][`crate::ReturnErrorCode::EcdsaRecoveryFailed]
+	fn ecdsa_to_eth_address(pubkey: &[u8; 33], output: &mut [u8; 20]) -> Result;
+
+	/// Stores the weight left into the supplied buffer.
+	///
+	/// Equivalent to the newer [`Self::gas_left_v1`] version but
+	/// works with *ref_time* Weight only.
+	fn gas_left(out: &mut &mut [u8]);
+
+	/// Stores the amount of weight left into the supplied buffer.
+	/// The data is encoded as Weight.
+	///
+	/// If the available space in `output` is less than the size of the value a trap is triggered.
+	///
+	/// # Parameters
+	///
+	/// - `output`: A reference to the output data buffer to write the weight left.
+	#[deprecated(note = "Unstable, use `gas_left` instead")]
+	fn gas_left_v1(output: &mut &mut [u8]);
+
+	/// Retrieve the value under the given key from storage.
+	///
+	/// This version is to be used with a fixed sized storage key. For runtimes supporting
+	/// transparent hashing, please use the newer version of this function.
+	fn get_storage(key: &[u8], output: &mut &mut [u8]) -> Result;
+
+	/// Retrieve the value under the given key from storage.
+	///
+	/// The key length must not exceed the maximum defined by the contracts module parameter.
+	///
+	/// # Parameters
+	/// - `key`: The storage key.
+	/// - `output`: A reference to the output data buffer to write the storage entry.
+	///
+	/// # Errors
+	///
+	/// [KeyNotFound][`crate::ReturnErrorCode::KeyNotFound]
+	fn get_storage_v1(key: &[u8], output: &mut &mut [u8]) -> Result;
+
+	hash_fn!(sha2_256, 32);
+	hash_fn!(keccak_256, 32);
+	hash_fn!(blake2_256, 32);
+	hash_fn!(blake2_128, 16);
+
+	/// Stores the input passed by the caller into the supplied buffer.
+	///
+	/// # Note
+	///
+	/// This function traps if:
+	/// - the input is larger than the available space.
+	/// - the input was previously forwarded by a [`call()`][`Self::call()`].
+	///
+	/// # Parameters
+	///
+	/// - `output`: A reference to the output data buffer to write the input data.
+	fn input(output: &mut &mut [u8]);
+
+	/// Instantiate a contract with the specified code hash.
+	#[deprecated(note = "Deprecated, use newer version instead")]
```

Comment:
Kind of obvious. We should state why it is deprecated instead: It is unsafe to not limit the PoV size. However, the only safe function is unstable :D

Reply:
yeah maybe we just scrape this one, this one became deprecated because re removed the deprecated  `code_hash_len` and `value_len` on the host side

Reply:
Yes sure we can remove it.